docs/getting_started.md

```diff
@@ -171,233 +208,453 @@ Also adding optimizer & model to this state has the advantage that we only need
+  state = train_step(state, batch) # get updated train state (which contains the updated parameters)
+  state = compute_metrics(state=state, batch=batch) # aggregate batch metrics
+
+  if (step+1) % num_steps_per_epoch == 0: # one training epoch has passed
+    for metric,value in state.metrics.compute().items(): # compute metrics
+      metrics_history[f'train_{metric}'].append(value) # record metrics
+    state = state.replace(metrics=state.metrics.empty()) # reset train_metrics for next training epoch
+
+    # Compute metrics on the test set after each training epoch
+    test_state = state
+    for test_batch in test_ds.as_numpy_iterator():
+      test_state = compute_metrics(state=test_state, batch=test_batch)
+
+    for metric,value in test_state.metrics.compute().items():
+      metrics_history[f'test_{metric}'].append(value)
+    
+    print(f"train epoch: {(step+1) // num_steps_per_epoch}, loss: {metrics_history['train_loss'][-1]}, accuracy: {metrics_history['train_accuracy'][-1] * 100}")
+    print(f"test epoch: {(step+1) // num_steps_per_epoch}, loss: {metrics_history['test_loss'][-1]}, accuracy: {metrics_history['test_accuracy'][-1] * 100}")
+```
 
-state = create_train_state(init_rng, learning_rate, momentum)
-del init_rng  # Must not be used anymore.
++++ {"id": "gfsecJzvzgCT"}
+
+## 13. Visualize metrics
+
+```{code-cell}
+---
+colab:
+  height: 353
+executionInfo:
+  elapsed: 346
+  status: ok
+  timestamp: 1671500871338
+  user:
+    displayName: Marcus Chiam
+    userId: '17531616275590396120'
+  user_tz: 300
+id: Zs5atiqIG9Kz
+outputId: 7cc6a375-83fe-4a29-fcac-9889212b8aed
+---
+# Plot loss and accuracy in subplots
+fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(15, 5))
+ax1.set_title('Loss')
+ax2.set_title('Accuracy')
+for dataset in ('train','test'):
+  ax1.plot(metrics_history[f'{dataset}_loss'], label=f'{dataset}_loss')
+  ax2.plot(metrics_history[f'{dataset}_accuracy'], label=f'{dataset}_accuracy')
+ax1.legend()
+ax2.legend()
+plt.show()
+plt.clf()
 ```
 
-+++ {"id": "UqNrWu7kIC9S"}
++++ {"id": "qQbKS0tV3sZ1"}
 
-## 14. Train and evaluate
+## 14. Inference on test set
 
-Once the training and testing is done after 10 epochs, the output should show that your model was able to achieve approximately 99% accuracy.
+Use the learned parameters to do model inference on the test set and visualize the images and their corresponding predicted labels.
 
 ```{code-cell}
-:id: 0nxgS5Z5IsT_
+---
+executionInfo:
+  elapsed: 98
+  status: ok
+  timestamp: 1671500871612
+  user:
+    displayName: Marcus Chiam
+    userId: '17531616275590396120'
+  user_tz: 300
+id: DFwxgBQf44ks
+---
+for test_batch in test_ds.as_numpy_iterator():
+  test_state = compute_metrics(state=test_state, batch=test_batch)
+  pred = state.apply_fn({'params': state.params}, test_batch['image']) # model inference
+  break # get only the first batch
+pred = pred.argmax(axis=1) # argmax the logits to get predicted labels
```

Comment:
Realistically we might want to create jitted function for inference e.g:
```suggestion
@jax.jit
def pred_step(state, batch):
  logits = state.apply_fn({'params': state.params}, test_batch['image'])
  return logits.argmax(axis=1) 

test_batch = test_ds.as_numpy_iterator().next()
pred = pred_step(state, test_batch)
```

Reply:
Even if we don't create `pred_step` its better to use:
```python
test_batch = test_ds.as_numpy_iterator().next()
```
instead of `break` in the loop.